aea/helpers/base.py

```diff
@@ -485,6 +485,33 @@ def recursive_update(
             to_update[key] = value
 
 
+def perform_dict_override(
```

Comment:
Method to perform the dict overrides, the reason I introduced this method is it's really hard to update the `recursive_update` method without breaking stuff because of this particular piece of code 

https://github.com/valory-xyz/open-aea/blob/bb1e6142cfa9992a9cd09264b78e8db221ba78c9/aea/helpers/base.py#L464-L466

This was the reason why the first time it was updating the recursive param with new key rather than replacing it with new value.